packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -134,6 +146,33 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
     columnsInitRef.current = true;
   }, [hasColumnsDefined, rows, nodeRuntime]);
 
+  React.useEffect(() => {
+    if (!nodeRuntime || rows.length <= 0 || idFieldInitRef.current) {
+      return;
+    }
+
+    if (rowIdFieldProp || Boolean(rows?.[0]?.id)) {
+      idFieldInitRef.current = true;
+      return;
+    }
+
+    const mappedRows = addIdField(rows);
+    nodeRuntime.setProp('rows', mappedRows);
+
+    idFieldInitRef.current = true;
+  }, [rowIdFieldProp, rows, nodeRuntime]);
+
+  const debouncedRowIdFieldProp = useDebounced(rowIdFieldProp, 300);
```

Comment:
Let's debounce this in the editor instead, before it updates the dom. No need to make the app rerender on every keystroke if we're going to debounce it in the end anyway

Reply:
Understood - I've added debouncing logic in the `StringPropEditor` so that the dom updates are debounced themselves. Reimplemented the logic since I couldn't find the scope to re-use `useDebounced` there.

Reply:
I've been thinking about this and we can probably get rid of debouncing altogether if we merge https://github.com/mui/mui-toolpad/pull/355